### ISSUE TITLE
Another case for "Cannot convert input parameter to type 'Microsoft.DurableTask.TaskOrchestrationContext' from type 'System.String'"

### ISSUE THREAD

**Issue by DL444** [microsoft/durabletask-dotnet#232]
Similar to #43, but this time it happens only with `FunctionsAutoRegisterGeneratedMetadataProvider` [enabled](https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#placeholders-preview).

## Minimal reproducible sample

*Function.cs*
```cs
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Http;
using Microsoft.DurableTask;
using Microsoft.DurableTask.Client;
using Microsoft.Extensions.Logging;

namespace DurableRepro
{
    public static class Function
    {
        [Function(nameof(Function))]
        public static async Task RunOrchestrator([OrchestrationTrigger] TaskOrchestrationContext context, List<string> input)
            => await Task.WhenAll(input.Select(x => context.CallActivityAsync(nameof(SayHello), x)));

        [Function(nameof(SayHello))]
        public static void SayHello([ActivityTrigger] string name, FunctionContext executionContext)
            => executionContext.GetLogger(nameof(SayHello)).LogInformation("Saying hello to {name}.", name);

        [Function("DurableRepro_HttpStart")]
        public static async Task<HttpResponseData> HttpStart(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req,
            [DurableClient] DurableTaskClient client,
            FunctionContext executionContext)
        {
            await client.ScheduleNewOrchestrationInstanceAsync(nameof(Function), new List<string>()
            {
                "Tokyo",
                "Seattle",
                "London",
            });
            return req.CreateResponse(System.Net.HttpStatusCode.OK);
        }
    }
}
```
*Program.cs*
```cs
using Microsoft.Extensions.Hosting;

var host = new HostBuilder()
    .ConfigureFunctionsWorkerDefaults()
    .Build();

host.Run();
```
*DurableRepro.csproj*
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
    <OutputType>Exe</OutputType>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <FunctionsEnableWorkerIndexing>True</FunctionsEnableWorkerIndexing>
    <FunctionsAutoRegisterGeneratedMetadataProvider>True</FunctionsAutoRegisterGeneratedMetadataProvider> <!-- Disable to circumvent issue -->
    <FunctionsEnableExecutorSourceGen>True</FunctionsEnableExecutorSourceGen>
  </PropertyGroup>
  <ItemGroup>
    <FrameworkReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.19.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" Version="1.0.4" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.1.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.15.1" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <Using Include="System.Threading.ExecutionContext" Alias="ExecutionContext" />
  </ItemGroup>
</Project>
```

## Observations
Triggering the client function either locally or deployed on Azure will throw the following exception when invoking orchestrator:
> Exception: Microsoft.Azure.Functions.Worker.FunctionInputConverterException: Error converting 1 input parameters for Function 'Function': Cannot convert input parameter 'context' to type 'Microsoft.DurableTask.TaskOrchestrationContext' from type 'System.String'. Error:System.Text.Json.JsonException: 'C' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0.

See [attached file](https://github.com/microsoft/durabletask-dotnet/files/13170752/exception.txt) for full stacktrace.

Disabling `FunctionsAutoRegisterGeneratedMetadataProvider` in project file circumvents the problem.

**Comment by jviau** [comment]
@DL444 thank you for the report. This was due to us using a case-sensitive comparison in our middleware. When using the code-generation route, the trigger type was `OrchestrationTrigger` (pascal case). We were checking for `orchestrationTrigger` (camel case). This caused our match to fail and thus input binding failed.

Good news is this issue was already identified and fixed, the fix is available in 1.1.0-preview.2. I have verified the repro you provided fails before upgrading and succeeds after.
